kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kms/aws/AwsKmsLocal.java

```diff
@@ -0,0 +1,218 @@
+
+    @Override
+    public boolean isAvailable() {
+        if (!isDeployed()) {
+            return false;
+        }
+        try (var output = new ByteArrayOutputStream();
+                var exec = kubeClient().getClient().pods()
+                        .inNamespace(deploymentNamespace)
+                        .withName(podName)
+                        .writingOutput(output)
+                        .exec("sh", "-c", AWS_LOCAL_CMD + " --version")) {
+            int exitCode = exec.exitCode().join();
+            return exitCode == 0 &&
+                    output.toString().toLowerCase().contains("aws-cli/");
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    /**
+     * Gets the installed version.
+     *
+     * @return the version
+     */
+    public String getLocalStackVersionInstalled() {
+        if (installedLocalStackVersion == null) {
+            URI url = getAwsUrl();
+            try (var output = new ByteArrayOutputStream();
+                    var error = new ByteArrayOutputStream();
+                    var exec = kubeClient().getClient().pods()
+                            .inNamespace(deploymentNamespace)
+                            .withName(podName)
+                            .writingOutput(output)
+                            .writingError(error)
+                            .exec("sh", "-c", "curl " + url + "/_" + LOCALSTACK_SERVICE_NAME + "/info")) {
+                int exitCode = exec.exitCode().join();
+                if (exitCode != 0) {
+                    throw new UnsupportedOperationException(error.toString());
+                }
+                // version returned with format: "3.5.1.dev:6e7ddd05e"
+                installedLocalStackVersion = new ObjectMapper().readTree(output.toString()).findValue("version").textValue().split(":")[0];
+            }
+            catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+
+        return installedLocalStackVersion;
+    }
+
+    @Override
+    public void deploy() {
+        LOGGER.info("Deploy AWS in {} namespace", deploymentNamespace);
+        if (isDeployed()) {
+            LOGGER.warn("Skipping AWS deployment. It is already deployed!");
+            return;
+        }
+
+        NamespaceUtils.createNamespaceWithWait(deploymentNamespace);
+        ResourceManager.helmClient().addRepository(LOCALSTACK_HELM_REPOSITORY_NAME, LOCALSTACK_HELM_REPOSITORY_URL);
+        ResourceManager.helmClient().namespace(deploymentNamespace).install(LOCALSTACK_HELM_CHART_NAME, LOCALSTACK_SERVICE_NAME,
+                Optional.of(Environment.LOCALSTACK_CHART_VERSION),
+                Optional.of(Path.of(TestUtils.getResourcesURI("helm_localstack_overrides.yaml"))),
+                Optional.empty());
+
+        this.podName = KafkaUtils.getPodNameByLabel(deploymentNamespace, "app.kubernetes.io/name", LOCALSTACK_SERVICE_NAME, Duration.ofSeconds(30));
```

Comment:
could we just use `--wait` flag to the Helm install?

Reply:
We are waiting for Helm installation already, adding `--wait` flag. But we need to be sure that pod is ready to be able to run whatever we need inside the pod. Waiting Helm install to install does not assure that the pod is already in `running` state.

Reply:
https://helm.sh/docs/intro/using_helm/#helpful-options-for-installupgraderollback

> Waits until all Pods are in a ready state....

Reply:
Yeap, I forgot to mention that I have checked and removed the `waitUntilPodRunning` call.